bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/commands/AddAnnotationCommandHandler.java

```diff
@@ -0,0 +1,101 @@
+package edu.kit.kastel.eclipse.grading.view.commands;
+
+import java.util.Optional;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jface.text.source.ISourceViewer;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.ui.texteditor.AbstractTextEditor;
+
+import edu.kit.kastel.eclipse.common.api.model.IMistakeType;
+import edu.kit.kastel.eclipse.common.view.utilities.AssessmentUtilities;
+import edu.kit.kastel.eclipse.grading.view.assessment.CustomButtonDialog;
+import edu.kit.kastel.eclipse.grading.view.controllers.AssessmentViewController;
+
+public class AddAnnotationCommandHandler extends AbstractHandler {
+	private final AssessmentViewController controller;
+	
+	public AddAnnotationCommandHandler(AssessmentViewController controller) {
+		this.controller = controller;
+	}
+	
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		if (this.controller.getAssessmentController() != null) {
+			var dialog = new AddAnnotationDialog(AssessmentUtilities.getWindowsShell(), this.controller);
+			//var dialog = new AddAnnotationFilteringDialog(AssessmentUtilities.getWindowsShell(), this.controller);
+			dialog.setBlockOnOpen(true);
+			dialog.create();
+			
+			var dialogPosition = getCaretPosition();
+			dialogPosition.x += 10;
```

Comment:
What does this number do and why?

Reply:
It offsets the dialog a few pixels to the right of the caret. I've found that this makes it easier to not loose track of the current code snippet when adding an annotation.

Reply:
Changed to a constant in the latest commit ;)